python/hsfs/core/arrow_flight_client.py

```diff
@@ -279,40 +280,61 @@ def _collect_featuregroups_features_and_filters_rec(self, query):
 
         return featuregroups, features, filters
 
-    def _filter_to_expression(self, filters, featuregroups):
+    def _filter_to_expression(self, filters, featuregroups, features):
```

Comment:
i don't doubt it works, i just think it would make sense to have features be part of the feature group (instead of a separate dictionary). Now the FG name is the link between the two dictionaries.

Also why not use existing FG and Feature classes?

Reply:
The problem this is PR is trying to solve is that some features don't have an associated featuregroup, I am establishing the association here.

Reply:
I agree though that it could be done prettier but in the end I need them as separate lists/dicts in exactly that format to send them to the fd backend.

Reply:
yeah, right now it looks like this
```
"featuregroups": {15: "test.fg_test_1"}
"features": {"test.fg_test_1": ["intt", "stringt"]}
```
but it might be better like this:
```
"featuregroups": [{"id": 15, "name": "test.fg_test_1", "features": ["intt", "stringt"]}]
```

Reply:
agree, but it's a bit out of scope for this PR